### PR TITLE
Delete collectd-mongodb monitor

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

Meant to temporarily resolve a CVE, we need to upgrade dependencies to resolve breaking changes and add testing, or move customers to use the native otel receivers.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,10 @@
   - Emit entity events only for matching receivers ([#4691](https://github.com/signalfx/splunk-otel-collector/pull/4691))
   - Remove `log_endpoints` config option ([#4692](https://github.com/signalfx/splunk-otel-collector/pull/4692))
 
+### 🚩 Deprecations 🚩
+
+- (Splunk) `collectd-mongodb`: Monitor has been removed to resolve CVE-2024-21506
```
Comment (crobert-1) [inline CHANGELOG.md:15]:
```suggestion
- (Splunk) `collectd-mongodb`: Monitor has been removed to resolve CVE-2024-21506 ([#4731](https://github.com/signalfx/splunk-otel-collector/pull/4731))
```

**Review by atoulme — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,10 @@
   - Emit entity events only for matching receivers ([#4691](https://github.com/signalfx/splunk-otel-collector/pull/4691))
   - Remove `log_endpoints` config option ([#4692](https://github.com/signalfx/splunk-otel-collector/pull/4692))
 
+### 🚩 Deprecations 🚩
+
+- (Splunk) `collectd-mongodb`: Monitor has been removed to resolve CVE-2024-21506 ([#4731](https://github.com/signalfx/splunk-otel-collector/pull/4731))
```
Comment (jeffreyc-splunk) [inline CHANGELOG.md:15]: Should this be a breaking change instead?

**Review by dmitryax — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (crobert-1) [reply CHANGELOG.md:15]: Sure. I had it under breaking change at first but thought `deprecation` might be slightly more accurate. I'll move it to breaking 👍

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,8 @@
   - Emit entity events only for matching receivers ([#4691](https://github.com/signalfx/splunk-otel-collector/pull/4691))
   - Remove `log_endpoints` config option ([#4692](https://github.com/signalfx/splunk-otel-collector/pull/4692))
 
+- (Splunk) `collectd-mongodb`: Monitor has been removed to resolve CVE-2024-21506 ([#4731](https://github.com/signalfx/splunk-otel-collector/pull/4731))
```
Comment (jinja2) [inline CHANGELOG.md:13]:
```suggestion
- (Splunk) `smartagent/collectd-mongodb`: Monitor has been removed to resolve CVE-2024-21506 ([#4731](https://github.com/signalfx/splunk-otel-collector/pull/4731))
```

**Review by jinja2 — APPROVED** [review]

**Pull request merged**